Utilities/OpenCorePackage/Script/Legacy/MountESP

```diff
@@ -0,0 +1,130 @@
+fi
+
+# check if target volume is a logical Volume instead of physical
+if [[ "$(echo $(LC_ALL=C diskutil list | grep -i 'Logical Volume' | \
+          awk '{print tolower($0)}'))" == *"logical volume"* ]]; then
+    # ok, we have a logical volume somewhere.. so that can assume that we can use "diskutil cs"
+    LC_ALL=C diskutil cs info $DiskDevice > /dev/null 2>&1
+    if [[ $? -eq 0 ]] ; then
+        # logical volumes does not have an EFI partition (or not suitable for us?)
+        echo "$DiskDevice is a logical volume"
+        # find the partition uuid
+        UUID=$(LC_ALL=C diskutil info "${DiskDevice}" 2>/dev/null | \
+                                      sed -n 's/.*artition UUID: *//p')
+        # with the partition uuid we can find the real disk in in diskutil list output
+        if [[ -n "$UUID" ]]; then
+            realDisk=$(LC_ALL=C diskutil list | \
+                grep -B 1 "$UUID" | \
+                grep -i 'logical volume' | awk '{print $4}' | \
+                sed -e 's/,//g' | sed -e 's/ //g')
+            if [[ -n "$realDisk" ]]; then
+                DiskDevice=$(LC_ALL=C diskutil info "${realDisk}" 2>/dev/null | \
+                                             sed -n 's/.*Part [oO]f Whole: *//p')
+            fi
+        fi
+    fi
+fi
+
+# check if target volume is APFS, and therefore part of an APFS container
+if [[ "$(echo $(LC_ALL=C diskutil list "$DiskDevice" | grep -i 'APFS Container Scheme' | \
+          awk '{print tolower($0)}'))" == *"apfs container scheme"* ]]; then
+    # ok, this disk is an APFS partition, extract physical store device
+    realDisk=$(LC_ALL=C diskutil list "$DiskDevice" 2>/dev/null | \
+        sed -n 's/.*Physical Store *//p')
+    echo Target volume "$1" on "$DiskDevice" is APFS on physical store "$realDisk"
+    DiskDevice=$(LC_ALL=C diskutil info "$realDisk" 2>/dev/null | \
+        sed -n 's/.*Part [oO]f Whole: *//p')
+fi
+
+# check one more time for RAID.
+if [[ ! -z "$(echo $(LC_ALL=C diskutil ar list | grep "$DiskDevice$"))" ]]; then
+  realDisk="$(diskutil ar list | grep -A 10 "${DiskDevice}$" | sed -n 's/Online//p' | head -1 | sed 's/[0-9]*[ ]*\([^ ]*\)[ ]*.*$/\1/g')"
+  # Find first online disk.
+  echo Target volume "$1" on "$DiskDevice" is RAID on physical store "$realDisk"
+  DiskDevice=$(LC_ALL=C diskutil info "$realDisk" 2>/dev/null | \
+  sed -n 's/.*Part [oO]f Whole: *//p')
+fi
+
+
+# echo "realDisk = $realDisk"
+# echo "UUID = $UUID"
+# echo "DiskDevice = $DiskDevice"
+
+# Check if the disk is a GPT disk
+disk_partition_scheme=$("$partutil" --show-partitionscheme "$DiskDevice")
+
+if [[ "$disk_partition_scheme" = "GUID_partition_scheme" ]]; then
+
+    plistbuddy='/usr/libexec/PlistBuddy'
+
+    # Get the index of the ESP device
+    index=$(LC_ALL=C /usr/sbin/gpt -r show "/dev/$DiskDevice" 2>/dev/null | \
+     awk 'toupper($7) == "C12A7328-F81F-11D2-BA4B-00A0C93EC93B" {print $3; exit}')
+    [[ -z "$index" ]] && index=1 # if not found use the index 1
```

Comment:
Such fallbacks are dangerous, both for accidentally mounting an undesired partition and for allowing ESPs with wrong GUIDs (lack of OS protection).